components/scream/src/share/field/field.hpp

```diff
@@ -135,7 +128,27 @@ class Field {
   // be reshaped to the desired layout before being used.
   template<typename DT, HostOrDevice HD = Device>
   get_view_type<uview_type<DT>,HD>
-  get_reshaped_view () const;
+  get_view () const;
+
+  // Allows to get the underlying 1d view. This can be dangerous, since
+  // this field may be a subfield of another one. In that case, the stored
+  // view would be that of the parent field, which would need to be read
+  // in a strided fashion. Furthermore, if the field has been allocated
+  // in a way that allows packing, there might be padding. However, the
+  // padding is inserted in the last physical dimension of the field,
+  // so when one sees the 1d view, the padding is interleaved in the array.
+  // For these reasons, this must be regarded as a *dangerous* power-user
+  // method. Do not use unless you know what you are doing.
+  // WARNING: this method throws if this field is a subfield. It does *NOT*
+  //          throw if there is padding, so careful how you access/use the view.
```

Comment:
Maybe this is a bad idea, but something to consider, anyway:

We could have an argument to this called `allow_padding`, set to `false` by default. If this argument is `true`, it would allow this method to be called on fields with padding. Otherwise it would throw.

Reply:
Something along that line might actually not be a bad idea.